### DISCUSSION TITLE
Whether the use of single-spa or qiankun for micro front terminal applications is supported?

### DISCUSSION THREAD

**Discussion by noob-ceng** [apache/superset#18489] [Ideas]
I had a problem with using the qiankun framework to make the project a child application and could not request a data interface。

qiankun: 2.4.2,
superset-ui/core: 0.17.10

**Comment by NoahLu1105** [comment]
i have  same problem  between qiankun and superset

**Comment by suddjian** [comment]
This is my first time learning about qiankun, but based on my introductory level of research it seems to me like the [Single Page App](https://github.com/apache/superset/projects/10) effort would need to be complete before this can be supported.

In the meantime, I'd love to learn more about your use case.

**Comment by shishiranshuman** [comment]
Is the Single Page App effort still on the horizon? The link share above seem to have expired.

**Reply by larsomic** [reply]
I'm not sure this would work great with single spa, although it would be nice. You would need to run the exact versions of react, react-dom and any other shared dependency. I think it would also depend on your use case. If you are trying to embed a part of the app like the sql-editor into your existing single-spa app it might be easier to just run the necessary parts and not the whole application.
